Source/WebKit/WebProcess/WebCoreSupport/WebNotificationClient.cpp

```diff
@@ -119,9 +123,18 @@ NotificationClient::Permission WebNotificationClient::checkPermission(ScriptExec
     callOnMainRunLoopAndWait([&resultPermission, origin = origin->data().toString().isolatedCopy()] {
         resultPermission = WebProcess::singleton().supplement<WebNotificationManager>()->policyForOrigin(origin);
     });
+
+    if (!m_notificationsRequesters.contains(context->securityOrigin()->data()) && resultPermission == NotificationClient::Permission::Denied)
```

Comment:
Maybe a comment here to explain why we're doing this. Like you did for the other code path.

Reply:
Fixed!